dask_expr/io/parquet.py

```diff
@@ -85,6 +87,128 @@ def _tokenize_fileinfo(fileinfo):
 _CACHED_PLAN_SIZE = 10
 _cached_plan = {}
 
+import pickle
+import weakref
+
+
+class FragmentWrapper:
+    _filesystems = weakref.WeakValueDictionary()
+
+    def __init__(self, fragment=None, file_size=None, fragment_packed=None) -> None:
+        """Wrap a pyarrow Fragment to only deserialize when needed."""
+        # https://github.com/apache/arrow/issues/40279
+        self._fragment = fragment
+        self._fragment_packed = fragment_packed
+        self._file_size = file_size
+        self._fs = None
+
+    def pack(self):
+        if self._fragment_packed is None:
+            self._fragment_packed = (
+                self._fragment.format,
+                (
+                    self._fragment.path
+                    if self._fragment.buffer is None
+                    else self._fragment.buffer
+                ),
+                pickle.dumps(self._fragment.filesystem),
```

Comment:
This is truly odd. When I keep the filesystem as is, I get a memory leak on the scheduler I'm afraid this is a genuine C++ memory leak but I couldn't reproduce it locally. I can just tell that from manual testing the leak vanished when I pickle this explicitly.